errors/pluginerrors.go

```diff
@@ -0,0 +1,159 @@
+	})
+
+	// ErrorCodeGetSubjectDescriptorFailure is returned when GetSubjectDescriptor
+	// API fails.
+	ErrorCodeGetSubjectDescriptorFailure = Register("errcode", ErrorDescriptor{
+		Value:       "GET_SUBJECT_DESCRIPTOR_FAILURE",
+		Message:     "get subject descriptor failure",
+		Description: `Generic error returned when the referrer store fails to get the subject descriptor.`,
+	})
+
+	// ErrorCodeGetReferenceManifestFailure is returned when GetReferenceManifest
+	// API fails.
+	ErrorCodeGetReferenceManifestFailure = Register("errcode", ErrorDescriptor{
+		Value:       "GET_REFERRER_MANIFEST_FAILURE",
+		Message:     "get reference manifest failure",
+		Description: `Generic error returned when the referrer store fails to get the reference manifest.`,
+	})
+
+	// ErrorCodeGetBlobContentFailure is returned when GetBlobContent API fails.
+	ErrorCodeGetBlobContentFailure = Register("errcode", ErrorDescriptor{
+		Value:       "GET_BLOB_CONTENT_FAILURE",
+		Message:     "get blob content failure",
+		Description: `Generic error returned when the referrer store fails to get the blob content.`,
+	})
+
+	// ErrorCodeReferrerStoreFailure is returned when a generic error happen in
+	// ReferrerStore.
+	ErrorCodeReferrerStoreFailure = Register("errcode", ErrorDescriptor{
+		Value:       "REFERRER_STORE_FAILURE",
+		Message:     "referrer store failure",
+		Description: `Generic error returned when the referrer store fails to perform an operation.`,
+	})
+
+	// ErrorCodeCreateRepositoryFailure is returned when Referrer Store fails to
+	// create a repository object.
+	ErrorCodeCreateRepositoryFailure = Register("errcode", ErrorDescriptor{
+		Value:       "CREATE_REPOSITORY_FAILURE",
+		Message:     "create repository failure",
+		Description: "Failed to create repository. Check the repository config for more details.",
+	})
+
+	// ErrorCodeRepositoryOperationFailure is returned when a repository
+	// operation fails.
+	ErrorCodeRepositoryOperationFailure = Register("errcode", ErrorDescriptor{
+		Value:       "REPOSITORY_OPERATION_FAILURE",
+		Message:     "repository operation failure",
+		Description: `The repository operation fails. Check the repository for more details.`,
+	})
+
+	// ErrorCodeManifestInvalid is returned if fetched manifest is invalid.
+	ErrorCodeManifestInvalid = Register("errcode", ErrorDescriptor{
+		Value:       "MANIFEST_INVALID",
+		Message:     "manifest invalid",
+		Description: `The manifest is invalid. Check the manifest for more details.`,
+	})
+
+	// ErrorCodeReferrersNotFound is returned if there is no ReferrerStore set.
+	ErrorCodeReferrersNotFound = Register("errcode", ErrorDescriptor{
+		Value:       "REFERRERS_NOT_FOUND",
+		Message:     "referrers not found",
+		Description: "No referrers are found. Check the artifact is configured correctly.",
+	})
+
+	// Generic errors happen in plugins
+
+	// ErrorCodePluginInitFailure is returned when it fails to initialize a plugin.
```

Comment:
suggestion for update, " returned when executor fails to initialize a plugin." better to be more descriptive about who can throw this error